migration/migrator/migrator.go

```diff
@@ -35,26 +34,40 @@ func ReadMigrationFiles(log hclog.Logger, migrationFiles embed.FS) (map[string][
 		err        error
 		migrations = make(map[string][]byte)
 	)
-	files, err := migrationFiles.ReadDir(migrationsEmbeddedDirectoryPath)
+
+	dirs, err := migrationFiles.ReadDir(migrationsEmbeddedDirectoryPath)
 	if err != nil {
 		log.Info("Provider doesn't define any migration files")
 		return migrations, nil
 	}
-	for _, m := range files {
-		f, err := migrationFiles.Open(path.Join(migrationsEmbeddedDirectoryPath, m.Name()))
-		if err != nil {
-			return nil, err
+	for _, d := range dirs {
+		if !d.IsDir() {
+			return nil, fmt.Errorf("bad migrations structure: missing dialect directories")
 		}
-		info, _ := m.Info()
-		if info.Size() == 0 {
-			migrations[m.Name()] = []byte("")
+		basePath := path.Join(migrationsEmbeddedDirectoryPath, d.Name())
+		files, err := migrationFiles.ReadDir(basePath)
```

Comment:
I don't like this timscale/x_vx_y_z.up.sql semantic, feels like we are hacking it in instead of solving it by reading the files saving it map[string]MigrationFiles.

Reply:
Done